proto/extensions.go

```diff
@@ -488,7 +488,7 @@ func SetExtension(pb Message, extension *ExtensionDesc, value interface{}) error
 	}
 	typ := reflect.TypeOf(extension.ExtensionType)
 	if typ != reflect.TypeOf(value) {
-		return errors.New("proto: bad extension value type")
+		return fmt.Errorf("proto: bad extension value type. want: %T, got: %T", extension.ExtensionType, value)
```

Comment:
Nit: Usually, we mention got before want.

Reply:
Done.